beaker/client/application_client.py

```diff
@@ -494,6 +494,9 @@ def call(
             **kwargs,
         )
 
+        if atc is None:
+            raise Exception("No ATC was built?")
```

Comment:
suggestion: avoid raising plain `Exception` instances, it makes it hard for callers to filter. In this case I think a `ValueError` would be appropriate here.

More concretely though, I believe you've added this for typing reasons - `add_method_call` is missing a return type annotation which should not include `None`, since it just takes  `atc: AtomicTransactionCompose` and returns it.